src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -2,22 +1,48 @@
+import { Dispatch, SetStateAction } from 'react';
+import { Filter } from '../../App';
+import { useDispatch } from 'react-redux';
+import { filterSlice } from '../../features/filter';
+
+type Props = {
+  filterType: string;
+  query: string;
+  onQuery: Dispatch<SetStateAction<string>>;
+  onResetQuery: () => void;
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  filterType,
+  query,
+  onQuery,
+  onResetQuery,
+}) => {
+  const dispatch = useDispatch();
 
-export const TodoFilter: React.FC = () => {
   return (
-    <form
-      className="field has-addons"
-      onSubmit={event => event.preventDefault()}
-    >
+    <form className="field has-addons">
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
-            <option value="all">All</option>
-            <option value="active">Active</option>
-            <option value="completed">Completed</option>
+          <select
+            data-cy="statusSelect"
+            value={filterType}
+            onChange={event => {
+              dispatch(filterSlice.actions.toggleFilter(event.target.value));
+            }}
+          >
+            <option value={Filter.All}>All</option>
+            <option value={Filter.Active}>Active</option>
+            <option value={Filter.Completed}>Completed</option>
           </select>
         </span>
       </p>
 
       <p className="control is-expanded has-icons-left has-icons-right">
         <input
+          value={query}
+          onChange={text => {
+            onQuery(text.target.value);
+            dispatch(filterSlice.actions.settingQuery(text.target.value));
```

Comment:
Verify that the `settingQuery` action in `filterSlice` is correctly defined to handle query updates. This action should update the state with the new query value.